mewarpx/mewarpx/mwxrun.py

```diff
@@ -49,50 +49,88 @@ def __init__(self):
+        ]
+        upper_boundary_conditions_parts = [
+            parts_boundary_xmax, parts_boundary_zmax
+        ]
+
         self.grid = picmi.Cartesian2DGrid(
             number_of_cells=[nx, nz],
             lower_bound=[xmin, zmin],
             upper_bound=[xmax, zmax],
-            lower_boundary_conditions=['periodic', 'dirichlet'],
-            upper_boundary_conditions=['periodic', 'dirichlet'],
-            lower_boundary_conditions_particles=['periodic', 'absorbing'],
-            upper_boundary_conditions_particles=['periodic', 'absorbing'],
-            warpx_max_grid_size=nz//max_grid_size
+            lower_boundary_conditions=lower_boundary_conditions,
+            upper_boundary_conditions=upper_boundary_conditions,
+            lower_boundary_conditions_particles=lower_boundary_conditions_parts,
+            upper_boundary_conditions_particles=upper_boundary_conditions_parts,
+            warpx_max_grid_size=nz//min_tiles
         )
 
         self._set_geom_str()
         self._set_grid_params()
 
-    def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
+    def init_timestep(self, DT=None, CFL_factor=None, V_grid=5):
         """Calculate timestep size based on grid data and CFL parameter
 
          Arguments:
-            V_anode (float): Vacuum bias of anode relative to cathode, V.
             DT (float): The dt of each step of the simulation, if not given it
                 will be calculated.
             CFL_factor (float): Multiplier to determine the actual timestep
                 given the CFL ratio. eg. ``dt = CFL_factor * CFL_ratio``.
             V_grid (float): The vaccum bias of highest-voltage grid relative to
-                the cathode V. If not defined a value of 5 is used, which
+                the cathode V. If not defined a value of 5 V is used, which
                 is a safe value for finding vmax in the case that the
-                electron velocities are only set by their temperature
+                electron velocities are only set by their temperature.
         """
         if self.grid is None:
-            raise ValueError("init_grid must be called before init_timestep!")
+            raise ValueError("init_grid must be called before init_timestep.")
 
         if DT is not None:
             self.dt = DT
             self.simulation.time_step_size = self.dt
             return self.dt
 
         if CFL_factor is None:
-            raise ValueError("No CFL_factor passed into init_timestep when DT is not defined!")
+            raise ValueError(
+                "Either CFL-factor or DT should be passed to init_timestep."
+            )
+        if V_grid == 0:
+            raise ValueError(
+                "V_grid must be non-zero to calculate timestep."
+            )
 
-        vmax = np.sqrt(
-               2*constants.e/constants.m_e * max(0.0, abs(V_anode), abs(V_grid))
-        )
+        vmax = np.sqrt(2*constants.e/constants.m_e * abs(V_grid))
```

Comment:
Do you think we should warn if say |V_grid| < 1? It looks like 0 will cause divide-by-0 error, but it makes me nervous that with thermal velocity not included in this calculation, it'll be incorrect for very low values of V_grid, which might be seen in eg a V_grid sweep.